lib/Frontend/FrontendInputsAndOutputs.cpp

```diff
@@ -284,10 +286,12 @@ void FrontendInputsAndOutputs::forEachInputProducingAMainOutputFile(
       : hasPrimaryInputs() ? forEachPrimaryInput(fn) : forEachInput(fn);
 }
 
-void FrontendInputsAndOutputs::setMainOutputs(
-    ArrayRef<std::string> outputFiles) {
-  assert(countOfInputsProducingMainOutputs() == outputFiles.size());
-  if (hasPrimaryInputs()) {
+void FrontendInputsAndOutputs::setMainAndSupplementaryOutputs(
+    ArrayRef<std::string> outputFiles,
+    SupplementaryOutputPaths supplementaryOutputs) {
+  if (outputFiles.empty())
+    ;
```

Comment:
Does this need to be handled explicitly at all? When is the assertion from before no longer valid?

Reply:
I just pushed a new commit that cleans up these corner cases. Check it out!